x/interchainstaking/keeper/redemptions.go

```diff
@@ -60,8 +60,11 @@ func (k *Keeper) processRedemptionForLsm(ctx sdk.Context, zone *types.Zone, send
 			TokenizedShareOwner: destination,
 		})
 	}
-	// add unallocated dust.
-	msgs[0].Amount = msgs[0].Amount.AddAmount(outstanding)
+	// Ensure there are messages to process
+	if len(msgs) > 0 {
```

Comment:
I think we don't need to check the `len(msgs)` here because it always > 0 if the code can run here, just need to add `//nolint:gosec` to pass the lint